cognite/client/_api/spaces.py

```diff
@@ -0,0 +1,190 @@
+
+        Args:
+            chunk_size (int, optional): Number of spaces to return in each chunk. Defaults to yielding one space a time.
+            limit (int, optional): Maximum number of events to return. Default to return all items.
+            partitions (int): Retrieve spaces in parallel using this number of workers. Also requires `limit=None` to be passed.
+
+        Yields:
+            Union[Space, SpaceList]: yields Space one by one if chunk_size is not specified, else SpaceList objects.
+        """
+        return self._list_generator(
+            list_cls=SpaceList,
+            resource_cls=Space,
+            method="GET",
+            chunk_size=chunk_size,
+            limit=limit,
+            partitions=partitions,
+        )
+
+    def __iter__(self) -> Iterator[Space]:
+        """Iterate over events
+
+        Fetches events as they are iterated over, so you keep a limited number of events in memory.
+
+        Yields:
+            Space: yields Spaces one by one.
+        """
+        return cast(Iterator[Space], self())
+
+    def retrieve(self, space: str) -> Optional[Space]:
+        """`Retrieve a single space by id. <https://docs.cognite.com/api/v1/#tag/Spaces/operation/bySpaceIdsSpaces>`_
+
+        Args:
+            space (str): Space ID
+
+        Returns:
+            Optional[Space]: Requested space or None if it does not exist.
+
+        Examples:
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> res = c.models.spaces.retrieve(space='mySpace')
+
+        """
+        identifiers = DataModelIdentifierSequence.load_spaces(spaces=space).as_singleton()
+        return self._retrieve_multiple(list_cls=SpaceList, resource_cls=Space, identifiers=identifiers)
+
+    def retrieve_multiple(
+        self,
+        spaces: Sequence[str],
+    ) -> SpaceList:
+        """`Retrieve multiple spaces by id. <https://docs.cognite.com/api/v1/#tag/Spaces/operation/bySpaceIdsSpaces>`_
+
+        Args:
+            spaces (Sequence[str]): Space IDs.
+
+        Returns:
+            SpaceList: The requested spaces.
+
+        Examples:
+
+            Get events by id::
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> res = c.models.spaces.retrieve_multiple(spaces=["MySpace", "MyAwesomeSpace", "MyOtherSpace"])
+
+        """
+        identifiers = DataModelIdentifierSequence.load_spaces(spaces=spaces)
+        return self._retrieve_multiple(list_cls=SpaceList, resource_cls=Space, identifiers=identifiers)
+
+    def list(
+        self,
+        partitions: int = None,
+        limit: int = LIST_LIMIT_DEFAULT,
+    ) -> SpaceList:
+        """`List events <https://docs.cognite.com/api/v1/#tag/Spaces/operation/listSpacesV3>`_
+
+        Args:
+            partitions (int): Retrieve events in parallel using this number of workers. Also requires `limit=None` to be passed.
```

Comment:
```suggestion
```

Reply:
Lots of other references to "events" in these docstrings.

Reply:
Sorry, forgot to do a search for them. Fixed now.